src/script/api/script_vehiclelist.cpp

```diff
@@ -107,11 +107,17 @@ ScriptVehicleList_Station::ScriptVehicleList_Station(StationID station_id)
 	EnforceDeityOrCompanyModeValid_Void();
 	if (!ScriptBaseStation::IsValidBaseStation(station_id)) return;
 
-	for (const Vehicle *v : Vehicle::Iterate()) {
-		if ((v->owner == ScriptObject::GetCompany() || ScriptCompanyMode::IsDeity()) && v->IsPrimaryVehicle()) {
-			for (const Order *order : v->Orders()) {
+	for (const OrderList *orderlist : OrderList::Iterate()) {
+		const Vehicle *v = orderlist->GetFirstSharedVehicle();
+		/* We assume all vehicles sharing an order list have the same owner. */
+		if ((v->owner == ScriptObject::GetCompany() || ScriptCompanyMode::IsDeity())) {
+			/* Vehicle is a candidate, search for a matching order. */
+			for (const Order *order = orderlist->GetFirstOrder(); order != nullptr; order = order->next) {
```

Comment:
```suggestion
			for (const Order *order = v->Orders()) {
```
Isn't this faster?

Reply:
Well, you tell me?

Reply:
It isn't, nevermind, sorry...
![image](https://github.com/OpenTTD/OpenTTD/assets/43006711/0115883b-a9ef-488c-b9c9-a09248954d36)